docs/reference/glossary.md

```diff
@@ -58,44 +86,67 @@ Layer 2 (L2) is a separate blockchain that increases transaction speed and throu
+
 ## Q
 
 ## R
 
-**Rollup**
+### Rollup
 
 A type of layer 2 scaling solution that batches multiple transactions and submits them to the Ethereum main chain in a single transaction. This allows for reductions in gas costs and increases in transaction throughput. There are optimistic and zero-knowledge rollups which use different security methods to offer these scalability gains.
 
+Rollup architecture is made up the following components:
+
+- **Rollup contract**: contract on the L1 stores rollup blocks, monitors state updates on the rollup, and tracks user deposits
+- **Off-chain VM**: computation and state storage on another virtual machine separate from the Ethereum Virtual Machine
+- **Aggregators/sequencers/operators/validators**: nodes that aggregate transactions, compress the underlying data, and publish the block on Ethereum
+- **Verifier contract**: contract on the L1 that verifies the validity proof
+
 ## S
 
-**Scaling**
+### Scaling
 
 The main goal of scalability is to increase transaction speed (faster finality), and transaction throughput (high transactions per second), without sacrificing decentralization or security.
 
-**Sidechain**
+### Sidechain
 
-A scaling solution that uses a separate chain with different, often faster, consensus rules. A bridge is needed to connect these sidechains to Mainnet. Rollups also use sidechains, but they operate in collaboration with Mainnet instead.
+A scaling solution that uses a separate chain with different, often faster, consensus rules. A bridge is needed to connect these sidechains to Mainnet. Rollups also use sidechains, but they operate in collaboration with Mainnet instead. Sidechains are NOT considered layer 2 solutions, because they do not leverage Ethereum for data availability and security.
 
 ## T
 
-**Transaction fee**
+### Transaction fee
 
 A fee you need to pay whenever you use the Ethereum network. Examples include sending funds from your wallet or a dapp interaction, like swapping tokens or buying a collectible. You can think of this like a service charge. This fee will change based on how busy the network is. This is because miners, the people responsible for processing your transaction, are likely to prioritize transactions with higher fees – so congestion forces the price up.
 
 ## U
 
 ## V
 
-**Validity proof**
+### Validity proof
 
 A security model for certain layer 2 solutions where, to increase speed, transactions are rolled up into batches and submitted to Ethereum in a single transaction. The transaction computation is done off-chain and then supplied to the main chain with a proof of their validity. This method increases the amount of transactions possible while maintaining security. Some rollups use fraud proofs.
```

Comment:
We need to update our style guide as regards the usage of "Layer 2", "layer 2", "L2", etc., but: we should at least be consistent. you use 'layer 2' here, and 'L2' further down. I would suggest you pick one and stick with it.